---
title: "Infinite Site Model"
output: rmarkdown::html_vignette
description: >
  How to enable/disable the infinite site model
vignette: >
  %\VignetteIndexEntry{Infinite Site Model}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options:
  markdown:
    wrap: 72
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

> *Disclaimer:* RACES/rRACES internally implements the
> probability distributions using the C++11 random number
> distribution classes. The standard does not specify their
> algorithms, and the class implementations are left free
> for the compiler. Thus, the simulation output depends on
> the compiler used to compile RACES, and because of that,
> the results reported in this article may differ from
> those obtained by the reader.

The mutation engine places mutation on the sampled cell genome according to the
[infinite sites model](https://en.wikipedia.org/wiki/Infinite_sites_model) by default.
In particular, any new mutations is placed on a locus whose context is mutation-free.

Let us consider the phylogenetic forest as build in `vignette("mutations")` and verify
whether it satisfies the infinite site model.


```{r, echo=FALSE,  results='hide', message=FALSE}
library(rRACES)

# set the seed of the random number generator
set.seed(0)

# create a simulation
sim <- SpatialSimulation()

sim$death_activation_level <- 20

# First mutant
sim$add_mutant(name = "A",
               epigenetic_rates = c("+-" = 0.01, "-+" = 0.01),
               growth_rates = c("+" = 0.1, "-" = 0.08),
               death_rates = c("+" = 0.1, "-" = 0.01))

sim$place_cell("A+", 500, 500)
sim$run_up_to_size("A+", 1000)

bbox_width <- 50
sim$sample_cells("S_1_1",
                 bottom_left = c(480, 480),
                 top_right = c(480 + bbox_width, 480 + bbox_width))

sim$sample_cells("S_1_2",
                 bottom_left = c(500, 500),
                 top_right = c(500 + bbox_width, 500 + bbox_width))

sim$run_up_to_time(sim$get_clock() + 15)

cell <- sim$choose_cell_in("A")

sim$add_mutant(name = "B",
               epigenetic_rates = c("+-" = 0.05, "-+" = 0.1),
               growth_rates = c("+" = 0.8, "-" = 0.3),
               death_rates = c("+" = 0.05, "-" = 0.05))

sim$mutate_progeny(cell, "B")

# let it grow more time units
sim$run_up_to_size("B+", 7000)

n_w <- n_h <- 25
ncells <- 0.9 * n_w * n_h

bbox <- sim$search_sample(c("A" = ncells), n_w, n_h)
sim$sample_cells("S_2_1", bbox$lower_corner, bbox$upper_corner)

bbox <- sim$search_sample(c("B" = ncells), n_w, n_h)
sim$sample_cells("S_2_2", bbox$lower_corner, bbox$upper_corner)

forest <- sim$get_samples_forest()

# save the samples forest in the file "samples_forest.sff"
forest$save("samples_forest.sff")

# building a mutation engine by using the "demo" setup
m_engine <- MutationEngine(setup_code = "demo")

# add the mutant "A" characterized by one driver SNV on the allele 1 of
# chromosome 22, one indel deletion on the same chromosome, and two CNAs:
# a deletion on the allele 1 of chromosome 22 and an amplification on a
# random allele of the same chromosome. The mutant has two epigenetic
# states and its species "A+" and "A-" have passenger SNV rates 1e-9 and
# 3e-8, respectively, passenger indel rates 1e-10 and 0, respectively,
# and passenger CNA rates 0 and 1e-11, respectively.
m_engine$add_mutant(mutant_name = "A",
                    passenger_rates = list("+" = c(SNV = 1e-9, indel = 1e-8),
                                           "-" = c(SNV = 3e-8, CNA = 1e-11)),
                    drivers = list(SNV("22", 10510210, "C", allele = 1),
                                   Mutation("22", 15220157, "GTTTTTTTT", "G"),
                                   CNA(type = "A", chr = "22",
                                       chr_pos = 10303470, len = 200000),
                                   CNA("D", "22", 5010000, 200000,
                                       allele = 1)))

# add the mutant "B" characterized by three driver SNVs on chromosome 22 (no
# CNA) and two epigenetic states. The first two SNVs are "NF2 R262", which
# can lay in any available allele, and "NF2 R221*", which instead must lay
# in the allele 1. The remaining SNV is specified by using the SNV function
# as done above. The species "B+" and "B-" have passenger SNV rates 8e-7
# and 5e-8, respectively, and CNA rates 0 for both species.
m_engine$add_mutant("B", list("+" = c(SNV = 8e-7), "-" = c(SNV = 5e-8)),
                    list(SNV("22", 12028576, "G"),         # the first SNV
                         "NF2 R262*",                      # the second SNV
                         list("NF2 R221*", allele = 1)))   # the third SNV


# add SNV and indel default exposures. This will be used from simulated time 0
# up to the successive exposure change.
m_engine$add_exposure(coefficients = c(SBS13 = 0.2, SBS1 = 0.8))
m_engine$add_exposure(c(ID2 = 0.6, ID13 = 0.2, ID21 = 0.2))

# add a new SNV exposure that will be used from simulated
# time 100 up to the successive exposure change.
m_engine$add_exposure(time = 100, c(SBS5 = 0.3, SBS2 = 0.2, SBS3 = 0.5))

m_engine$add_exposure(time = 120, c(SBS5 = 0.3, SBS2 = 0.2, SBS3 = 0.5,
                                    ID1 = 0.8, ID9 = 0.2))

subjects <- m_engine$get_germline_subjects()

m_engine$set_germline_subject(subjects[2, "sample"])

samples_forest <- load_samples_forest("samples_forest.sff")

# place mutations on the sample forest assuming 1000 pre-neoplastic SNVs and
# 500 indels
phylo_forest <- m_engine$place_mutations(samples_forest, 1000, 500)

detach("package:rRACES", unload=TRUE)
```

```{r}
library(rRACES)
library(dplyr)

# this function verifies whether any mutation arised in
# two unrelated cells represented in a phylogenetic forest
test_infinite_sites_model <- function(phylo_forest) {
  # extract non-germinal SNVs that appear multiple times either
  # in the same cell or different cells
  snvs <- phylo_forest$get_sampled_cell_mutations() %>%
    filter(class != "germinal", .data$type == "SNV") %>%
    count(.data$chr, .data$chr_pos, .data$ref, .data$alt) %>%
    filter(n > 1)

  # search for an SNV that independently occurred in two unrelated cells
  first_occurrences <- c()
  row <- 1
  while (length(first_occurrences) < 2 && row <= nrow(snvs)) {
    snv <- SNV(snvs[row, "chr"], snvs[row, "chr_pos"],
               ref = snvs[row, "ref"], alt = snvs[row, "alt"])

    first_occurrences <- phylo_forest$get_first_occurrences(snv)
    row <- row + 1
  }

  # if the last handled SNV independently occurred in two unrelated
  # cells at least
  if (length(first_occurrences) >= 2) {

    # print a message containing the two cells
    paste0("SNV('", snv$get_chromosome(), "'',",
           snv$get_position_in_chromosome(), ",'", snv$get_ref(),
           "','", snv$get_alt(),
           "') independently arises in cells ", first_occurrences[1],
           " and ", first_occurrences[2])
  } else {
    print("Every mutation arises exclusively in one cell")
  }
}

# test whether the infinite sites conditions hold in the built forest
test_infinite_sites_model(phylo_forest)
```

This behavior can be changed by using the mutation engine property
`MutationEngine$infinite_sites_model()`. This property is a Boolean flag
that enable/disable the infinite sites model.

```{r}
# establish whether the infinite sites model is used
m_engine$infinite_sites_model

# disable it
m_engine$infinite_sites_model <- FALSE
```

When the infinite sites model is disabled,
`MutationEngine$place_mutations()` may place two mutations
in the same locus of different alleles of the same genome
or the same mutation in the same locus of one allele
of the genomes of two cells that are not each other ancestors.

```{r}
# test whether the infinite sites model is enable
m_engine$infinite_sites_model

# place the mutations on the same samples forest used above
phylo_forest2 <- m_engine$place_mutations(samples_forest, 1000, 500)

# test whether the infinite sites conditions hold in the new forest
test_infinite_sites_model(phylo_forest2)
```

