<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Generating phylogenetic forests — MutationEngine • rRACES</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Generating phylogenetic forests — MutationEngine"><meta name="description" content="A mutation engine can label every node of a descendants
forest by mutations and produce a consistent phylogenetic forest.
This function downloads and sets up the data
requires by a mutation engine. Finally, it builds mutation
engine itself."><meta property="og:description" content="A mutation engine can label every node of a descendants
forest by mutations and produce a consistent phylogenetic forest.
This function downloads and sets up the data
requires by a mutation engine. Finally, it builds mutation
engine itself."><meta property="og:image" content="https://caravagnalab.github.io/rRACES/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">rRACES</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/rRACES.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/tissue_simulation.html">Tissue simulation</a></li>
    <li><a class="dropdown-item" href="../articles/sampling.html">Tissue sampling</a></li>
    <li><a class="dropdown-item" href="../articles/mutations.html">Simulating Mutations</a></li>
    <li><a class="dropdown-item" href="../articles/sequencing.html">Sequencing Simulation</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">More articles...</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/rRACES/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Generating phylogenetic forests</h1>
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/rRACES/R/RcppExports.R" class="external-link"><code>R/RcppExports.R</code></a></small>
      <div class="d-none name"><code>MutationEngine.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A mutation engine can label every node of a descendants
forest by mutations and produce a consistent phylogenetic forest.</p>
<p>This function downloads and sets up the data
requires by a mutation engine. Finally, it builds mutation
engine itself.</p>
    </div>


    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-setup-code">setup_code<a class="anchor" aria-label="anchor" href="#arg-setup-code"></a></dt>
<dd><p>The set-up code (alternative to <code>directory</code>).</p></dd>


<dt id="arg-directory">directory<a class="anchor" aria-label="anchor" href="#arg-directory"></a></dt>
<dd><p>The set-up directory (alternative to <code>setup_code</code>).</p></dd>


<dt id="arg-reference-src">reference_src<a class="anchor" aria-label="anchor" href="#arg-reference-src"></a></dt>
<dd><p>The reference genome path or URL (mandatory when
<code>directory</code> is provided).</p></dd>


<dt id="arg-sbs-signatures-src">SBS_signatures_src<a class="anchor" aria-label="anchor" href="#arg-sbs-signatures-src"></a></dt>
<dd><p>The SBS signature file path or URL (mandatory
when <code>directory</code> is provided).</p></dd>


<dt id="arg-indel-signatures-src">indel_signatures_src<a class="anchor" aria-label="anchor" href="#arg-indel-signatures-src"></a></dt>
<dd><p>The indel signature file path or URL (mandatory
when <code>directory</code> is provided).</p></dd>


<dt id="arg-drivers-src">drivers_src<a class="anchor" aria-label="anchor" href="#arg-drivers-src"></a></dt>
<dd><p>The driver mutation file path or URL (mandatory when
<code>directory</code> is provided).</p></dd>


<dt id="arg-passenger-cnas-src">passenger_CNAs_src<a class="anchor" aria-label="anchor" href="#arg-passenger-cnas-src"></a></dt>
<dd><p>The passenger CNAs file path or URL (mandatory
when <code>directory</code> is provided).</p></dd>


<dt id="arg-germline-src">germline_src<a class="anchor" aria-label="anchor" href="#arg-germline-src"></a></dt>
<dd><p>The germline directory path or URL (mandatory when
<code>directory</code> is provided).</p></dd>


<dt id="arg-germline-subject">germline_subject<a class="anchor" aria-label="anchor" href="#arg-germline-subject"></a></dt>
<dd><p>The germline subject (optional).</p></dd>


<dt id="arg-context-sampling">context_sampling<a class="anchor" aria-label="anchor" href="#arg-context-sampling"></a></dt>
<dd><p>The number of reference contexts per context in
the index (optional: default value is 100).</p></dd>


<dt id="arg-max-index-size">max_index_size<a class="anchor" aria-label="anchor" href="#arg-max-index-size"></a></dt>
<dd><p>The maximum size of an admitted indel and, as a
consequence, the maximum size of a motif stored in the repeated
sequence index (optional: default value is 50).</p></dd>


<dt id="arg-max-repetition-storage">max_repetition_storage<a class="anchor" aria-label="anchor" href="#arg-max-repetition-storage"></a></dt>
<dd><p>The maximum number of repetitions per type
stored in the repeated sequence index (optional: default value is
500000).</p></dd>


<dt id="arg-tumour-type">tumour_type<a class="anchor" aria-label="anchor" href="#arg-tumour-type"></a></dt>
<dd><p>The type of tumour. This is currently used to select
the admissible passenger CNAs. If any passenger CNA in the dataset is
admissible, use the the empty string <code>""</code> (optional: default value is
<code>""</code>).</p></dd>


<dt id="arg-tumour-study">tumour_study<a class="anchor" aria-label="anchor" href="#arg-tumour-study"></a></dt>
<dd><p>The nationality code of the tumour study. This is
used to select the admissible passenger CNAs. If any tumor study in
the dataset is admissible, use the the empty string <code>""</code> (optional:
default value is <code>""</code>).</p></dd>


<dt id="arg-avoid-homozygous-losses">avoid_homozygous_losses<a class="anchor" aria-label="anchor" href="#arg-avoid-homozygous-losses"></a></dt>
<dd><p>An optional Boolean flag to avoid
homozygous losses. When set to <code>TRUE</code>, passenger CNAs will be
exclusively applied to regions covered by two alleles at least.
(default: TRUE).</p></dd>


<dt id="arg-quiet">quiet<a class="anchor" aria-label="anchor" href="#arg-quiet"></a></dt>
<dd><p>An optional Boolean flag to avoid the progress bar
(default: FALSE).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A mutation engine object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The mutations are randomly generated according to three
factors:
- the mutational rates of the species involved in the descendants
forest
- the genotycal characterization of the mutants involved in the
descendants forest, i.e., the SVNs and CNAs characterizing the
mutant genotypes
- the SBS signature coefficients active along the species
simulation</p>
<p>These data are provided to a mutation engine by using the methods
<code><a href="MutationEngine-cash-add_exposure.html">MutationEngine$add_exposure()</a></code> and <code><a href="MutationEngine-cash-add_exposure.html">MutationEngine$add_exposure()</a></code>
These data are provided by means of the <code><a href="MutationEngine-cash-add_mutant.html">MutationEngine$add_mutant()</a></code>.</p>
<p>The construction of a <code>MutationEngine</code> object requires a reference
sequence and an SBS file which are downloaded from the Internet.
After the download a context index of the reference sequence is then
automatically built. Thess processes may take time depending on the
size of the reference sequence. Because of this, the downloaded files
together with the context index are saved in a directory on the disk
and they are available for successive <code>MutationEngine</code> constructions.</p>
<p>There are two building modalities: the first one is more
general, but it requires to specify all the data sources; the
second one adopts some pre-set configurations, but it is more
convient than the former in many cases.</p>
<p>The first building modality requires to specify the directory in
which the data must be saved, the path or URL of the reference
sequence, the SBS file, the driver SNVs file, the passenger CNAs
file, and the germline data directory through the parameters
<code>directory</code>, <code>reference_src</code>, <code>SBS_src</code>, <code>drivers_src</code>,
<code>passenger_CNAs_src</code>, and <code>germline_src</code>, respectively.</p>
<p>The second building modality exclusively requires a set-up code
(parameter <code>setup_code</code>). The list of supported set-up codes can
be obtained by using the function <code><a href="get_mutation_engine_codes.html">get_mutation_engine_codes()</a></code>.</p>
<p>The number of context sampling is an optional parameter that allows
sampling the reference contexts while building the context index.
This parameter, which is set to 100 by default, specifies how many
occurences of the same context must be identified before adding
one of them to the context index. The larger the number of context
sampling, the larger the context index. On the other side, the
lower the number of context sampling, the lower the number of sites
in the reference genome that can be affected by simulated
mutations.</p>
<p>If the parameters of a mutation engine construction match those
of a previous construction, then the corresponding reference
sequence, the SBS file, and the previously built context index
are loaded from the set-up directory avoiding further
computations.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="get_mutation_engine_codes.html">get_mutation_engine_codes()</a></code> provides a list of the supported
set-up codes.</p>
<p><code><a href="MutationEngine-cash-get_germline_subjects.html">MutationEngine$get_germline_subjects()</a></code> to get the available
germline subjects; <code><a href="MutationEngine-cash-set_germline_subject.html">MutationEngine$set_germline_subject()</a></code> to set the
active germline subject; <code><a href="MutationEngine-cash-get_active_germline.html">MutationEngine$get_active_germline()</a></code> to get
the active germline subject.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># set the reference and SBS URLs</span></span></span>
<span class="r-in"><span><span class="va">reference_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://ftp.ensembl.org/pub/grch37/release-111/"</span>,</span></span>
<span class="r-in"><span>                        <span class="st">"fasta/homo_sapiens/dna/Homo_sapiens.GRCh37."</span>,</span></span>
<span class="r-in"><span>                        <span class="st">"dna.chromosome.22.fa.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sbs_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://cancer.sanger.ac.uk/signatures/documents/"</span>,</span></span>
<span class="r-in"><span>                  <span class="st">"2123/COSMIC_v3.4_SBS_GRCh37.txt"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">indel_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://cancer.sanger.ac.uk/signatures/documents/"</span>,</span></span>
<span class="r-in"><span>                    <span class="st">"2121/COSMIC_v3.4_ID_GRCh37.txt"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">drivers_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/"</span>,</span></span>
<span class="r-in"><span>                      <span class="st">"caravagnalab/rRACES/main/inst/extdata/"</span>,</span></span>
<span class="r-in"><span>                      <span class="st">"driver_mutations_hg19.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">passenger_CNAs_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/"</span>,</span></span>
<span class="r-in"><span>                             <span class="st">"caravagnalab/rRACES/main/inst/extdata/"</span>,</span></span>
<span class="r-in"><span>                             <span class="st">"passenger_CNAs_hg19.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">germline_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/13166780/files/"</span>,</span></span>
<span class="r-in"><span>                       <span class="st">"germline_data_demo.tar.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># build a mutation engine and save the required files into the</span></span></span>
<span class="r-in"><span><span class="co"># directory "Test". The `drivers_url` parameter is optional, but</span></span></span>
<span class="r-in"><span><span class="co"># it is suggested to avoid passenger mutations on driver loci.</span></span></span>
<span class="r-in"><span><span class="va">m_engine</span> <span class="op">&lt;-</span> <span class="fu">MutationEngine</span><span class="op">(</span>directory <span class="op">=</span> <span class="st">"Test"</span>,</span></span>
<span class="r-in"><span>                                  reference_src <span class="op">=</span> <span class="va">reference_url</span>,</span></span>
<span class="r-in"><span>                                  SBS_signatures_src <span class="op">=</span> <span class="va">sbs_url</span>,</span></span>
<span class="r-in"><span>                                  indel_signatures_src <span class="op">=</span> <span class="va">indel_url</span>,</span></span>
<span class="r-in"><span>                                  drivers_src <span class="op">=</span> <span class="va">drivers_url</span>,</span></span>
<span class="r-in"><span>                                  passenger_CNAs_src <span class="op">=</span> <span class="va">passenger_CNAs_url</span>,</span></span>
<span class="r-in"><span>                                  germline_src <span class="op">=</span> <span class="va">germline_url</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Downloading reference genome...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reference genome downloaded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Decompressing reference file...done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Downloading SBS file...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SBS file downloaded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Downloading indel file...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> indel file downloaded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Downloading driver mutation file...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Driver mutation file downloaded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Downloading passenger CNAs file...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Passenger CNAs file downloaded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Downloading germline mutations...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Germline mutations downloaded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Building context index...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Processing chr. 22                                        [█████████████████-----------------------] 40% [00m:01s] Processing chr. 22                                       [█████████████████████████████████-------] 81% [00m:02s] Processing chr. 22                                       [████████████████████████████████████████] 100% [00m:02s] Context index built                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Saving context index                                      [████████████████████████████████████████] 100% [00m:00s] Context index saved                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Building repeated sequence index...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:00s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:03s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:04s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:06s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:07s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:09s] Processing chr. 22                                        [█---------------------------------------] 0% [00m:13s] Processing chr. 22                                        [████████████████████████████████████████] 100% [00m:13s] RS index built                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Saving RS index                                           [█---------------------------------------] 0% [00m:00s] Saving RS index                                           [█████████████████████████---------------] 60% [00m:02s] Saving RS index                                         done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [████████████████████████████████████████] 100% [00m:02s] RS index saved                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading germline                                          [████████████████████████████████████████] 100% [00m:00s] Germline loaded                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Saving germline                                           [████████████████████████████████████████] 100% [00m:00s] Germline saved                                         </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># if the parameters of a mutation engine construction match those of a</span></span></span>
<span class="r-in"><span><span class="co"># previous construction, then the corresponding reference sequence,</span></span></span>
<span class="r-in"><span><span class="co"># the SBS file, and the previously built context index are loaded from</span></span></span>
<span class="r-in"><span><span class="co"># the set-up directory avoiding further computations.</span></span></span>
<span class="r-in"><span><span class="va">m_engine</span> <span class="op">&lt;-</span> <span class="fu">MutationEngine</span><span class="op">(</span><span class="st">"Test"</span>, <span class="va">reference_url</span>, <span class="va">sbs_url</span>,</span></span>
<span class="r-in"><span>                                  <span class="va">indel_url</span>, <span class="va">drivers_url</span>,</span></span>
<span class="r-in"><span>                                  <span class="va">passenger_CNAs_url</span>, <span class="va">germline_url</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading context index                                     [████████████████████████████████████████] 100% [00m:00s] Context index loaded                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading RS index                                          [████████████████------------------------] 38% [00m:01s] Loading RS index                                         [███████████████████████████████---------] 75% [00m:02s] Loading RS index                                         [████████████████████████████████████████] 100% [00m:02s] RS index loaded                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading germline                                          [████████████████████████████████████████] 100% [00m:00s] Germline loaded                                        </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># if the `context_sampling` parameter changes, a new context index is</span></span></span>
<span class="r-in"><span><span class="co"># built, while neither the reference sequence nor the SBS file are</span></span></span>
<span class="r-in"><span><span class="co"># downloaded again.</span></span></span>
<span class="r-in"><span><span class="va">m_engine</span> <span class="op">&lt;-</span> <span class="fu">MutationEngine</span><span class="op">(</span><span class="st">"Test"</span>, <span class="va">reference_url</span>, <span class="va">sbs_url</span>,</span></span>
<span class="r-in"><span>                                  <span class="va">indel_url</span>, <span class="va">drivers_url</span>,</span></span>
<span class="r-in"><span>                                  <span class="va">passenger_CNAs_url</span>, <span class="va">germline_url</span>,</span></span>
<span class="r-in"><span>                                  context_sampling <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Building context index...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Processing chr. 22                                        [█████████████████-----------------------] 40% [00m:01s] Processing chr. 22                                       [█████████████████████████████████-------] 81% [00m:02s] Processing chr. 22                                       [████████████████████████████████████████] 100% [00m:02s] Context index built                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Saving context index                                      [████████████████████████████████████████] 100% [00m:00s] Context index saved                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading RS index                                          [██████████████--------------------------] 34% [00m:01s] Loading RS index                                         [█████████████████████████████-----------] 70% [00m:02s] Loading RS index                                         [████████████████████████████████████████] 100% [00m:02s] RS index loaded                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading germline                                          [████████████████████████████████████████] 100% [00m:00s] Germline loaded                                        </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># a futher contruction with the same parameters avoids both downloads</span></span></span>
<span class="r-in"><span><span class="co"># and context index construction.</span></span></span>
<span class="r-in"><span><span class="va">m_engine</span> <span class="op">&lt;-</span> <span class="fu">MutationEngine</span><span class="op">(</span><span class="st">"Test"</span>, <span class="va">reference_url</span>, <span class="va">sbs_url</span>,</span></span>
<span class="r-in"><span>                                  <span class="va">indel_url</span>, <span class="va">drivers_url</span>,</span></span>
<span class="r-in"><span>                                  <span class="va">passenger_CNAs_url</span>, <span class="va">germline_url</span>,</span></span>
<span class="r-in"><span>                                  context_sampling <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading context index                                     [████████████████████████████████████████] 100% [00m:00s] Context index loaded                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading RS index                                          [████████████████------------------------] 38% [00m:01s] Loading RS index                                         [███████████████████████████████---------] 75% [00m:02s] Loading RS index                                         [████████████████████████████████████████] 100% [00m:02s] RS index loaded                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading germline                                          [████████████████████████████████████████] 100% [00m:00s] Germline loaded                                        </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">m_engine</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MutationEngine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Passenger rates</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Driver mutations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Timed Exposure</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    SBS Timed Exposures</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    indel Timed Exposures</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the parameters `directory`, `reference_src`, `SBS_src`, `drivers_src`,</span></span></span>
<span class="r-in"><span><span class="co"># `passenger_CNAs_src`, and `germline_src` can be avoided by providing</span></span></span>
<span class="r-in"><span><span class="co"># the `setup_code` parameter. The set-up code `demo` is provided among</span></span></span>
<span class="r-in"><span><span class="co"># those available for testing purpose.</span></span></span>
<span class="r-in"><span><span class="va">m_engine</span> <span class="op">&lt;-</span> <span class="fu">MutationEngine</span><span class="op">(</span>setup_code <span class="op">=</span> <span class="st">"demo"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading context index                                     [████████████████████████████████████████] 100% [00m:00s] Context index loaded                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading RS index                                          [██████████████--------------------------] 34% [00m:01s] Loading RS index                                         [█████████████████████████---------------] 61% [00m:02s] Loading RS index                                         [████████████████████████████████████████] 98% [00m:03s] Loading RS index                                         [████████████████████████████████████████] 100% [00m:03s] RS index loaded                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading germline                                          [████████████████████████████████████████] 100% [00m:00s] Germline loaded                                        </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the `context_sampling` can be used also when a pre-defined set-up</span></span></span>
<span class="r-in"><span><span class="co"># configuration is adopted.</span></span></span>
<span class="r-in"><span><span class="va">m_engine</span> <span class="op">&lt;-</span> <span class="fu">MutationEngine</span><span class="op">(</span>setup_code <span class="op">=</span> <span class="st">"demo"</span>,</span></span>
<span class="r-in"><span>                                  context_sampling <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Building context index...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Processing chr. 22                                        [█████████████████-----------------------] 40% [00m:01s] Processing chr. 22                                       [█████████████████████████████████-------] 81% [00m:02s] Processing chr. 22                                       [████████████████████████████████████████] 100% [00m:02s] Context index built                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Saving context index                                      [████████████████████████████████████████] 100% [00m:00s] Context index saved                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading RS index                                          [███████████████-------------------------] 35% [00m:01s] Loading RS index                                         [█████████████████████████████-----------] 71% [00m:02s] Loading RS index                                         [████████████████████████████████████████] 100% [00m:02s] RS index loaded                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [█---------------------------------------] 0% [00m:00s] Loading germline                                          [████████████████████████████████████████] 100% [00m:00s] Germline loaded                                        </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">m_engine</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MutationEngine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Passenger rates</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Driver mutations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Timed Exposure</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    SBS Timed Exposures</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    indel Timed Exposures</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove the "Test" and "demo" directories</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"Test"</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"demo"</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alberto Casagrande, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

