<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>The phylogenetic forest of cells in samples — PhylogeneticForest • rRACES</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The phylogenetic forest of cells in samples — PhylogeneticForest"><meta name="description" content="This class represents the phylogenetic forest of the
cells sampled during the computation."><meta property="og:description" content="This class represents the phylogenetic forest of the
cells sampled during the computation."><meta property="og:image" content="https://caravagnalab.github.io/rRACES/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">rRACES</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/rRACES.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/tissue_simulation.html">Tissue simulation</a></li>
    <li><a class="dropdown-item" href="../articles/sampling.html">Tissue sampling</a></li>
    <li><a class="dropdown-item" href="../articles/mutations.html">Simulating Mutations</a></li>
    <li><a class="dropdown-item" href="../articles/sequencing.html">Sequencing Simulation</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">More articles...</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/rRACES/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>The phylogenetic forest of cells in samples</h1>
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/rRACES/R/RcppExports.R" class="external-link"><code>R/RcppExports.R</code></a></small>
      <div class="d-none name"><code>PhylogeneticForest.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This class represents the phylogenetic forest of the
cells sampled during the computation.</p>
    </div>


    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The leaves of his forest are the sampled cells.
This class is analoguous to the class <a href="SamplesForest.html">SamplesForest</a>,
but each node is labelled with the mutations occuring
for the first time on the cell represented by the node
itself. Moreover each leaf is also associated with the
genome mutations occurring in the corresponding cell.</p>
    </div>
    <div class="section level2">
    <h2 id="fields">Fields<a class="anchor" aria-label="anchor" href="#fields"></a></h2>


<dl><dt><code>get_coalescent_cells</code></dt>
<dd><p>Retrieve most recent common ancestors</p><ul><li><p><em>Parameter:</em> <code>cell_ids</code> - The list of the identifiers of the
cells whose most recent common ancestors are aimed (optional).</p></li>
<li><p><em>Return:</em> A dataframe representing, for each of the identified
cells, the identified (column <code>cell_id</code>), whenever the
node is not a root, the ancestor identifier (column
<code>ancestor</code>), whenever the node was sampled, i.e., it is
one of the forest leaves, the name of the sample
containing the node, (column <code>sample</code>), the mutant
(column <code>mutant</code>), the epistate (column <code>epistate</code>),
and the birth time (column <code>birth_time</code>).</p></li>
</ul></dd>


<dt><code>get_first_occurrences</code></dt>
<dd><p>Gets the identifier of the cell in which a
mutation occurs for the first time</p><ul><li><p><em>Parameter:</em> <code>mutation</code> - A mutation being a
SNV, a indel, or a CNA.</p></li>
<li><p><em>Return:</em> The identifier of the cell in which a mutation
occurs for the first time.</p></li>
</ul></dd>


<dt><code>get_germline_mutations</code></dt>
<dd><p>Gets the germinal SNVs and indels</p><ul><li><p><em>Return:</em> A dataframe reporting <code>chr</code> (i.e., the
chromosome), <code>chr_pos</code>" (i.e., the position in the chromosome),
<code>allele</code> (in which the SNV occurs), <code>ref</code>, <code>alt</code>, <code>type</code> (i.e., either
<code>"SNV"</code> or <code>"indel"</code>) and <code>class</code> (i.e., <code>"germinal"</code>).</p></li>
</ul></dd>


<dt><code>get_germline_subject</code></dt>
<dd><p>Gets the germline subject name</p><ul><li><p><em>Return:</em> The name of the subject whose germline is used.</p></li>
</ul></dd>


<dt><code>get_nodes</code></dt>
<dd><p>Get the forest nodes</p><ul><li><p><em>Return:</em> A dataframe representing, for each node
in the forest, the identified (column <code>id</code>),
whenever the node is not a root, the ancestor
identifier (column <code>ancestor</code>), whenever the node
was sampled, i.e., it is one of the forest
leaves, the name of the sample containing the
node, (column <code>sample</code>), the mutant (column
<code>mutant</code>), the epistate (column <code>epistate</code>),
and the birth time (column <code>birth_time</code>).</p></li>
</ul></dd>


<dt><code>get_sampled_cell_CNAs</code></dt>
<dd><p>Gets the CNAs of the sampled cells</p><ul><li><p><em>Returns:</em> A dataframe reporting <code>cell_id</code>, <code>type</code> (<code>"A"</code> for
amplifications and <code>"D"</code> for deletions), <code>chr</code>, <code>begin</code>
(i.e., the first CNA locus in the chromosome), <code>end</code> (i.e., the
last CNA locus in the chromosome), <code>allele</code>, <code>src allele</code>
(the allele origin for amplifications, <code>NA</code> for deletions), and
<code>class</code> (i.e., <code>"driver"</code>, <code>"passenger"</code>, <code>"germinal"</code> or
<code>"preneoplastic"</code>).</p></li>
</ul></dd>


<dt><code>get_sampled_cell_mutations</code></dt>
<dd><p>Gets the SNVs and the indels of the
sampled cells</p><ul><li><p><em>Returns:</em> A dataframe reporting <code>cell_id</code>, <code>chr</code>, (i.e., the
mutation chromosome), <code>begin</code> (i.e., position in the chromosome),
<code>allele</code> (in which the SNV occurs), <code>ref</code>, <code>alt</code>, <code>type</code> (i.e., either
<code>"SNV"</code> or <code>"indel"</code>), <code>cause</code>, and <code>class</code> (i.e., <code>"driver"</code>,
<code>"passenger"</code>, <code>"germinal"</code> or <code>"preneoplastic"</code>) for each mutation
in the sampled cell genomes.</p></li>
</ul></dd>


<dt><code>get_samples_info</code></dt>
<dd><p>Retrieve information about the samples</p><ul><li><p><em>Returns:</em> A dataframe containing, for each sample collected
during the simulation, the columns "<code>name</code>", "<code>time</code>", "<code>ymin</code>",
"<code>xmin</code>", "<code>ymax</code>", "<code>xmax</code>", "<code>tumour_cells</code>", and
"<code>tumour_cells_in_bbox</code>". The columns "<code>ymin</code>", "<code>xmin</code>", "<code>ymax</code>",
"<code>xmax</code>" report the boundaries of the sample bounding box, while
"<code>tumour_cells</code>" and "<code>tumour_cells_in_bbox</code>" are the number of
tumour cells in the sample and in the bounding box, respectively.</p></li>
</ul></dd>


<dt><code>get_species_info</code></dt>
<dd><p>Gets the species data</p><ul><li><p><em>Returns:</em> A dataframe reporting <code>mutant</code> and <code>epistate</code>
for each registered species.</p></li>
</ul></dd>


<dt><code>get_sticks</code></dt>
<dd><p>Compute the forest sticks</p><ul><li><p><em>Returns:</em> The list of the forest sticks. Each stick is represented as
the list of cell identifiers labelling the nodes in the stick
from the higher to the deeper in the forest.</p></li>
</ul></dd>


<dt><code>get_subforest_for</code></dt>
<dd><p>Build a subforest using as leaves some of the original samples</p><ul><li><p><em>Parameter:</em> <code>sample_names</code> - The names of the samples whose cells will be used
as leaves of the new forest.</p></li>
<li><p><em>Returns:</em> A samples forest built on the samples mentioned in <code>sample_names</code>.</p></li>
</ul></dd>


<dt><code>get_absolute_chromosome_positions</code></dt>
<dd><p>Get the absolute chromosome positions</p><ul><li><p><em>Returns:</em> A dataframe reporting the name (column "<code>chr</code>"), the length
(column "<code>length</code>"), the initial absolute position (column "<code>from</code>"),
and the final absolute position (column "<code>to</code>") of each chromosome.</p></li>
</ul></dd>


<dt><code>save</code></dt>
<dd><p>Save a phylogenetic forest in a file</p><ul><li><p><em>Parameter:</em> <code>filename</code> - The path of the file in which the phylogenetic
forest must be saved.</p></li>
</ul></dd>


</dl></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alberto Casagrande, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

